src/DssSpell.t.sol

```diff
@@ -543,4 +546,198 @@ contract DssSpellTest is DssSpellTestBase {
+        address flashKiller = chainLog.getAddress("FLASH_KILLER");
+
+        assertEq(vat.wards(address(flashCurrent)), 1);
+        assertEq(vat.wards(address(flashLegacy)), 1);
+
+        assertEq(flashLegacy.max(), 250 * MILLION * WAD);
+        assertEq(flashCurrent.max(), 250 * MILLION * WAD);
+        assertEq(flashLegacy.wards(pauseProxy), 1);
+        assertEq(flashLegacy.wards(flashKiller), 1);
+        assertEq(flashLegacy.wards(address(esm)), 1);
+
+        _vote(address(spell));
+        _scheduleWaitAndCast(address(spell));
+        assertTrue(spell.done());
+
+        assertEq(vat.wards(address(flashCurrent)), 1);
+        assertEq(vat.wards(address(flashLegacy)), 0);
+
+        assertEq(flashLegacy.max(), 0);
+        assertEq(flashCurrent.max(), 500 * MILLION * WAD);
+        assertEq(flashLegacy.wards(pauseProxy), 0);
+        assertEq(flashLegacy.wards(flashKiller), 0);
+        assertEq(flashLegacy.wards(address(esm)), 0);
+
+        vm.expectRevert(abi.encodePacked("dss-chain-log/invalid-key"));
+        chainLog.getAddress("MCD_FLASH_LEGACY");
+
+        vm.expectRevert(abi.encodePacked("dss-chain-log/invalid-key"));
+        chainLog.getAddress("FLASH_KILLER");
+    }
+
+    // Test to ensure test of flash mint is working before the spell is cast...
+    function testFlashWorksBeforeSpell() public {
+        //_vote(address(spell));
+        //_scheduleWaitAndCast(address(spell));
+        //assertTrue(spell.done());
+
+        uint256 vowDai = vat.dai(address(vow));
+
+        // Give ourselves tokens for repayment in the callbacks
+        _giveTokens(address(dai), 1_000 * WAD);
+
+        FlashAbstract flash = FlashAbstract(addr.addr("MCD_FLASH_LEGACY"));
+        assertEq(flash.vat(), address(vat));
+        assertEq(flash.daiJoin(), address(daiJoin));
+        assertEq(flash.dai(), address(dai));
+        assertEq(flash.vow(), address(vow));
+        assertEq(flash.max(), 250 * MILLION * WAD);
+        assertEq(flash.toll(), 0);
+        assertEq(flash.maxFlashLoan(address(dai)), 250 * MILLION * WAD);
+        assertEq(flash.flashFee(address(dai), 1 * MILLION * WAD), 0);
+        flash.flashLoan(address(this), address(dai), 1 * MILLION * WAD, "");
+        flash.vatDaiFlashLoan(address(this), 1 * MILLION * RAD, "");
+        assertEq(vat.sin(address(flash)), 0);
+        assertEq(vat.dai(address(flash)), 0);
+        flash.accrue();
+        assertEq(vat.dai(address(flash)), 0);
+        assertEq(vat.dai(address(vow)), vowDai);
+    }
+
+    // Test to ensure 1 Dai flash mint fails after spell
+    function testFlashFailsAfterSpell() public {
+        _vote(address(spell));
+        _scheduleWaitAndCast(address(spell));
+        assertTrue(spell.done());
+
+        // Give ourselves tokens for repayment in the callbacks
+        _giveTokens(address(dai), 1_000 * WAD);
+
+        FlashAbstract flash = FlashAbstract(addr.addr("MCD_FLASH_LEGACY"));
```

Comment:
Might be worth thinking about when we remove addresses from test configs after we've removed them from the chainlog.

Reply:
i was thinking this would happen with the next spell so we don't forget